packages/analytics/analytics-chart/src/components/chart-types/StackedBarChart.vue

```diff
@@ -339,38 +332,30 @@ const numLabels = computed(() => {
   return (props.chartData.labels && props.chartData.labels.length) || 0
 })
 
-const chartWidth = computed(() => {
-  let value: number | string = MIN_CHART_WIDTH
-
-  if (canvas.value) {
-    value = canvas.value.width
-
-    if (canvas.value && props.chartData?.labels && props.chartData?.labels.length > MAX_BARS_VERTICAL && !isHorizontal.value) {
-      const numLabels = props.chartData.labels.length
+const chartWidth = computed<string>(() => {
+  if (canvas.value && props.chartData?.labels && props.chartData?.labels.length > MAX_BARS_VERTICAL && !isHorizontal.value) {
+    const numLabels = props.chartData.labels.length
 
-      const baseWidth = canvas.value.offsetWidth
-      const preferredChartWidth = baseWidth + ((numLabels - MAX_BARS_VERTICAL) * MIN_BAR_WIDTH)
+    const baseWidth = canvas.value.offsetWidth
+    const preferredChartWidth = baseWidth + ((numLabels - MAX_BARS_VERTICAL) * MIN_BAR_WIDTH)
 
-      value = `${preferredChartWidth}px`
-    }
+    return `${preferredChartWidth}px`
   }
 
-  return value
+  return DEFAULT_CHART_WIDTH
 })
 
 const chartHeight = computed(() => {
-  let chartHeight = parseInt(props.height, 10)
-  if (numLabels.value && isHorizontal.value) {
+  if (canvas.value && props.chartData?.labels && props.chartData?.labels.length > MAX_BARS_HORIZONTAL && isHorizontal.value) {
+    const numLabels = props.chartData.labels.length
+
+    const baseHeight = canvas.value.offsetHeight
+    const preferredChartHeight = baseHeight + ((numLabels - MAX_BARS_HORIZONTAL) * MIN_BAR_WIDTH)
```

Comment:
Why is this calculation changing?  Is it possible to implement this without changing the calculation?

In the past, we ran into issues with trying to do `baseHeight + <dynamic height>` because it's harder to calculate the dynamic part of the height -- for example, here the overall height may end up negative if there is not much canvas and only 1 bar.  That's why the old calculation relied on `Math.max`.  If we change it, we're going to need to do a lot more testing to make sure it behaves properly with varying numbers of bars.

Reply:
It can't be negative cause this code only gets execuded if numLabels is greater than `MAX_BARS_HORIZONTAL`

Reply:
Also, just making it more consistent with the width calculation

Reply:
nvm, moving to the old calculation for both the width and height computed props